src/hotspot/share/code/compiledMethod.cpp

```diff
@@ -687,12 +687,15 @@ address CompiledMethod::continuation_for_implicit_exception(address pc, bool for
     ResourceMark rm(thread);
     CodeBlob* cb = CodeCache::find_blob(pc);
     assert(cb != nullptr && cb == this, "");
-    ttyLocker ttyl;
-    tty->print_cr("implicit exception happened at " INTPTR_FORMAT, p2i(pc));
-    print();
-    method()->print_codes();
-    print_code();
-    print_pcs();
+
+    // Keep tty output consistent. To avoid ttyLocker, we buffer in stream, and print all at once.
+    stringStream ss;
+    ss.print_cr("implicit exception happened at " INTPTR_FORMAT, p2i(pc));
+    print_on(&ss);
+    method()->print_codes_on(&ss);
+    print_code_on(&ss);
+    print_pcs_on(&ss);
+    tty->print("%s", ss.as_string()); // print all at once
```

Comment:
It seems like these ttyLocker changes should be checked in as a different cleanup, ie removing ttyLocker is a really good thing.  Can you make these changes a separate patch?

Reply:
Yes, I filed the RFE:
[JDK-8324129](https://bugs.openjdk.org/browse/JDK-8324129) C2: Remove some ttyLocker usages in preparation for JDK-8306767
https://github.com/openjdk/jdk/pull/17486